src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,161 @@
+        userWithValidData.setLogin("user");
+        userWithValidData.setAge(21);
+
+        userWithValidDateSameLogin = new User();
+        userWithValidDateSameLogin.setLogin("user");
+        userWithValidDateSameLogin.setPassword("123456958");
+        userWithValidDateSameLogin.setAge(25);
+
+        userWithInvalidPassword = new User();
+        userWithInvalidPassword.setLogin("user4");
+        userWithInvalidPassword.setPassword("123");
+        userWithInvalidPassword.setAge(23);
+
+        userWithValidData1 = new User();
+        userWithValidData1.setLogin("user8");
+        userWithValidData1.setPassword("1235874589");
+        userWithValidData1.setAge(29);
+
+        userWithValidData2 = new User();
+        userWithValidData2.setLogin("user6");
+        userWithValidData2.setPassword("587884894986");
+        userWithValidData2.setAge(31);
+
+        userWithValidDateSameLogin2 = new User();
+        userWithValidDateSameLogin2.setLogin("user8");
+        userWithValidDateSameLogin2.setPassword("87456985");
+        userWithValidDateSameLogin2.setAge(42);
+
+        userWithNullAge = new User();
+        userWithNullAge.setLogin("user7");
+        userWithNullAge.setPassword("123456879");
+        userWithNullAge.setAge(null);
+
+        userWithNullLogin = new User();
+        userWithNullLogin.setLogin(null);
+        userWithNullLogin.setPassword("12387456879");
+        userWithNullLogin.setAge(31);
+
+        userWithNullPassword = new User();
+        userWithNullPassword.setLogin("user9");
+        userWithNullPassword.setPassword(null);
+        userWithNullPassword.setAge(28);
+
+    }
+
+    @Test
+    @Order(1)
+    void register_invalidAge_NotOk() {
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(userWithInvalidAge));
+    }
+
+    @Test
+    @Order(2)
+    void register_validData_Ok() {
+        User user = registrationService.register(userWithValidData);
+        assertEquals(userWithValidData, user);
+        assertEquals(1, userList.size());
+    }
+
+    @Test
+    @Order(3)
+    void register_sameLogin_NotOk() {
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(userWithValidDateSameLogin));
+    }
+
+    @Test
+    @Order(4)
+    void register_invalidPassword_NotOk() {
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(userWithInvalidPassword));
+    }
+
+    @Test
+    @Order(5)
+    void register_validData1_Ok() {
+        User user = registrationService.register(userWithValidData1);
+        assertEquals(userWithValidData1, user);
+        assertEquals(2, userList.size());
+    }
+
+    @Test
+    @Order(6)
+    void register_validData2_Ok() {
+        User user = registrationService.register(userWithValidData2);
+        assertEquals(userWithValidData2, user);
+        assertEquals(3, userList.size());
+    }
```

Comment:
```suggestion
    @Test
    @Order(5)
    void register_validData1_Ok() {
        User user = registrationService.register(userWithValidData1);
        assertEquals(userWithValidData1, user);
        assertEquals(2, userList.size());
    }

    @Test
    @Order(6)
    void register_validData2_Ok() {
        User user = registrationService.register(userWithValidData2);
        assertEquals(userWithValidData2, user);
        assertEquals(3, userList.size());
    }
```

No need to do the same tests for the same functionality. If you want to test multiple users, just add it to one test.